### PR TITLE
Removed types in MakeStruct IR node.

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

File: src/main/scala/is/hail/expr/ir/Copy.scala [hunk 1]
```diff
@@ -61,9 +61,8 @@ object Copy {
       case ArrayFold(_, _, accumName, valueName, _, typ) =>
         val IndexedSeq(a, zero, body) = children
         ArrayFold(a, zero, accumName, valueName, body, typ)
-      case MakeStruct(fields) =>
-        assert(fields.length == children.length)
-        MakeStruct(fields.zip(children).map { case ((n, t, _), v) => (n, t, v) })
+      case MakeStruct(fields, _) =>
+        MakeStruct(fields.zip(children).map { case ((n, _), a) => (n, a) })
```
Comment (danking) [inline src/main/scala/is/hail/expr/ir/Copy.scala]: if children is the wrong length this won't fail, which seems bad

File: src/main/scala/is/hail/expr/ir/IR.scala [hunk 2]
```diff
@@ -33,7 +33,7 @@ case class ApplyBinaryPrimOp(op: BinaryOp, l: IR, r: IR, var typ: Type = null) e
 case class ApplyUnaryPrimOp(op: UnaryOp, x: IR, var typ: Type = null) extends IR
 
 case class MakeArray(args: Array[IR], var typ: TArray = null) extends IR {
-  override def toString(): String = s"MakeArray(${args: IndexedSeq[IR]}, $typ)"
+  override def toString: String = s"MakeArray(${ args.toIndexedSeq }, $typ)"
```
Comment (danking) [inline src/main/scala/is/hail/expr/ir/IR.scala]: `.toIndexedSeq` allocates a Vector, which I guess doesn't much matter for printing the IR, but seems unnecessarily bad.

File: src/main/scala/is/hail/utils/richUtils/RichArray.scala [hunk 3]
```diff
@@ -3,4 +3,6 @@ package is.hail.utils.richUtils
 
 class RichArray[T](val a: Array[T]) extends AnyVal {
   def index: Map[T, Int] = a.zipWithIndex.toMap
+
+  def toFastIndexedSeq: IndexedSeq[T] = a
```
Comment (danking) [inline src/main/scala/is/hail/utils/richUtils/RichArray.scala]: I like this.

Comment (cseed) [reply src/main/scala/is/hail/utils/richUtils/RichArray.scala]: Yeah, don't know why we didn't add something like this sooner.

**Pull request merged**